src/pudl/workspace/setup.py

```diff
@@ -4,310 +4,120 @@
 import pathlib
 import shutil
 from pathlib import Path
-from typing import IO
 
-import yaml
-from dotenv import load_dotenv
+from pydantic import BaseSettings
 
 import pudl.logging_helpers
 
 logger = pudl.logging_helpers.get_logger(__name__)
 
 
-def get_defaults(
+def set_path_overrides(
     input_dir: str | None = None,
     output_dir: str | None = None,
-    yaml_file: IO | None = None,
-    default_pudl_yaml: Path | None = Path.home() / ".pudl.yml",
-) -> dict[str, str]:
-    """Derive PUDL workspace paths from specified input/output directories.
+) -> None:
+    """Set PUDL_INPUT and/or PUDL_OUTPUT env variables.
 
-    Determines input/output directory locations from YAML, then overrides with
-    env vars, then overrides with keywords passed in.
-
-    Input/output workspace roots can be the same directories.
+    Args:
+        input_dir: if set, overrides PUDL_INPUT env variable.
+        output_dir: if set, overrides PUDL_OUTPUT env variable.
+    """
+    if input_dir:
+        os.environ["PUDL_INPUT"] = input_dir
+    if output_dir:
+        os.environ["PUDL_OUTPUT"] = input_dir
```

Comment:
In general what's the idea behind using this global override of the environment variables (will it persist in the user's environment?) rather than creating a new `PudlPaths` object with arguments that override whatever the environment variables are within the scope of the use of that object?

It seems like the `BaseSettings` class is already designed to allow overriding the defaults via arguments when initializing the class, the `.env` file, or environment variables.

For the CLIs would it be better to just tell users "Set your environment variables or use .env" rather than having the arguments?  And then in the tests, since we want them to be isolatable from the user's environment, use a `PudlPaths` fixture that's been created with the appropriate test-only paths?

Reply:
If we can drop the support for command-line flags that would simplify things a bit.

re/ overrides - it seems to me that in many cases, the path configuration was accessed via `get_defaults()` which implies that it should have globally consistent value. I would further also assume that when you set the overrides, it should be okay to have it set to this value until the end of the process or until someone else sets up another override. 

Now that I think about it, it might be possible that some unit/integration tests assume that they run in a hermetic environment where inputs/outputs are isolated in between tests that run in the same process. But, given the above widespread use of `get_defaults()` I'm not sure if this behavior can be safe.

I generally like the idea of being able to get the paths simply by accessing `PudlPaths()` w/o having to pass this config around. For something that is stable/unchanging and globally consistent, this seems like a straightforward choice. Even for unit tests, we could consider simply setting the ENV variables in the `tox.ini` or via some other mechanism at the startup time.